pkg/i2gw/ingress2gateway.go

```diff
@@ -136,53 +140,53 @@ func GetSupportedProviders() []string {
 	return supportedProviders
 }
 
-// MergeGatewayResources accept multiple GatewayResources and create a unique Resource struct
-// built as follows:
+// MergeGatewayIRs accepts multiple IRs and creatse a unique IR struct built
+// as follows:
 //   - GatewayClasses, *Routes, and ReferenceGrants are grouped into the same maps
 //   - Gateways may have the same NamespaceName even if they come from different
 //     ingresses, as they have a their GatewayClass' name as name. For this reason,
 //     if there are mutiple gateways named the same, their listeners are merged into
 //     a unique Gateway.
 //
 // This behavior is likely to change after https://github.com/kubernetes-sigs/gateway-api/pull/1863 takes place.
-func MergeGatewayResources(gatewayResources ...GatewayResources) (GatewayResources, field.ErrorList) {
-	mergedGatewayResources := GatewayResources{
-		Gateways:        make(map[types.NamespacedName]gatewayv1.Gateway),
+func MergeIRs(irs ...IR) (IR, field.ErrorList) {
+	mergedIRs := IR{
+		Gateways:        make(map[types.NamespacedName]GatewayContext),
 		GatewayClasses:  make(map[types.NamespacedName]gatewayv1.GatewayClass),
-		HTTPRoutes:      make(map[types.NamespacedName]gatewayv1.HTTPRoute),
+		HTTPRoutes:      make(map[types.NamespacedName]HTTPRouteContext),
 		TLSRoutes:       make(map[types.NamespacedName]gatewayv1alpha2.TLSRoute),
 		TCPRoutes:       make(map[types.NamespacedName]gatewayv1alpha2.TCPRoute),
 		UDPRoutes:       make(map[types.NamespacedName]gatewayv1alpha2.UDPRoute),
 		ReferenceGrants: make(map[types.NamespacedName]gatewayv1beta1.ReferenceGrant),
 	}
 	var errs field.ErrorList
-	mergedGatewayResources.Gateways, errs = mergeGateways(gatewayResources)
+	mergedIRs.Gateways, errs = mergeGateways(irs)
 	if len(errs) > 0 {
-		return GatewayResources{}, errs
+		return IR{}, errs
 	}
-	for _, gr := range gatewayResources {
-		maps.Copy(mergedGatewayResources.GatewayClasses, gr.GatewayClasses)
-		maps.Copy(mergedGatewayResources.HTTPRoutes, gr.HTTPRoutes)
-		maps.Copy(mergedGatewayResources.TLSRoutes, gr.TLSRoutes)
-		maps.Copy(mergedGatewayResources.TCPRoutes, gr.TCPRoutes)
-		maps.Copy(mergedGatewayResources.UDPRoutes, gr.UDPRoutes)
-		maps.Copy(mergedGatewayResources.ReferenceGrants, gr.ReferenceGrants)
+	for _, gr := range irs {
+		maps.Copy(mergedIRs.GatewayClasses, gr.GatewayClasses)
+		maps.Copy(mergedIRs.HTTPRoutes, gr.HTTPRoutes)
+		maps.Copy(mergedIRs.TLSRoutes, gr.TLSRoutes)
+		maps.Copy(mergedIRs.TCPRoutes, gr.TCPRoutes)
+		maps.Copy(mergedIRs.UDPRoutes, gr.UDPRoutes)
+		maps.Copy(mergedIRs.ReferenceGrants, gr.ReferenceGrants)
 	}
-	return mergedGatewayResources, errs
+	return mergedIRs, errs
 }
 
-func mergeGateways(gatewaResources []GatewayResources) (map[types.NamespacedName]gatewayv1.Gateway, field.ErrorList) {
-	newGateways := map[types.NamespacedName]gatewayv1.Gateway{}
+func mergeGateways(irs []IR) (map[types.NamespacedName]GatewayContext, field.ErrorList) {
```

Comment:
Updated for GatewayContext. We don't currently perform any operations on HTTPRouteContext other than putting them together. There is no merge on HTTPRoutes.

Reply:
before, we had `MergeGatewayResources`, right?
This would have merged anything under GatewayResources, including Gateways, GatewayRoutes, HTTPRoutes, etc. (we can see this here on the left panel)

My question is how `maps.Copy(mergedIRs.HTTPRoutes, gr.HTTPRoutes)` would work given that HTTPRouteContext has nested HTTPRouteIR field with its own nested fields. 
Maybe maps.copy is enough for us, idk.. but something to check, I am not sure it will work out of the box

Reply:
They should work. I've tried to run tests on functions using this helper, and they passed with expected merged results.

Reply:
Well thats because you have nothing on the http IR yet. similarly to the logic you employ for the GatewayIR merging in https://github.com/kubernetes-sigs/ingress2gateway/pull/188/commits/db033246488ac4489fe484046baf3c7f6ecb7ace you would have to add logic to merge the HTTPIRs and ServiceIR, maps.copy() would not cover that.